### ISSUE TITLE
how to build dependency with a single command, when using package_revision_mode

### ISSUE THREAD

**Issue by dvorak0** [conan-io/conan#5709]
I am using `package_revision_mode` in my packages. Let's say the dependency graph to is `A->B->C`. All of the three packages need to be built from source. I expect `conan create A` to trigger building B and C automatically, which is not the case.

I check the docs and see:
> Given that the package ID of consumers depends on the package revision PREV of the dependencies, when one of the upstream dependencies doesn't have a package revision yet (for example it is going to be built from sources, so its PREV cannot be determined yet), the consumers package ID will be unknown and marked as such. These dependency graphs cannot be built in a single invocation, because they are intended for CI systems, in which a package creation/built is called for each package in the graph.

Why the procedure cannot be done in **a single invocation**. What is suggested to do?

I also found a solution: use `conan` info to get build order, and jenkins use `multibuild` to do this. It's not convenient at all.

**Comment by jgsogo** [comment]
You can explore the `build` argument to `conan create`, it accepts different values and also patterns so you can choose what to build:

```bash
  -b [BUILD], --build [BUILD]

    Optional, use it to choose if you want to build from sources:

    --build            Build all from sources, do not use binary packages.
    --build=never      Never build, use binary packages or fail if a binary package is not found.
    --build=missing    Build from code if a binary package is not found.
    --build=cascade    Will build from code all the nodes with some dependency being built
                       (for any reason). Can be used together with any other build policy.
                       Useful to make sure that any new change introduced in a dependency is
                       incorporated by building again the package.
    --build=outdated   Build from code if the binary is not built with the current recipe or
                       when missing binary package.
    --build=[pattern]  Build always these packages from source, but never build the others.
                       Allows multiple --build parameters. 'pattern' is a fnmatch file pattern
                       of a package reference.

    Default behavior: If you don't specify anything, it will be similar to '--build=never', but
    package recipes can override it with their 'build_policy' attribute in the conanfile.py.

```

If you want to always build from sources, use `conan create A --build`

**Comment by dvorak0** [comment]
Thanks jgsogo. I tried `build=missing`, but it still needs multiple runs.

`conan create A --build=missing` do build C, but not build B, reporting "B: package_ID_unknown"。

**Comment by jgsogo** [comment]
Hi, @dvorak0. Yes, you are right, it looks like there is no way to trigger a build of several nodes using `package_revision_mode`. I've tried with different `--build` arguments with no success.

I'm quite sure that the problem is Conan not being able to compute the package_id of the dependent nodes because it depends on the compiled binary hash (which is not known in advance). Conan should recompute it after building the requirement, but it is not the case. I'm tagging this as a _feature request_, although I'm not sure if we will need a different value for the argument, or it is something to add to the existing ones.

**Comment by dvorak0** [comment]
@jgsogo Thanks! Looking forward to the implementation of the recomputation.

**Comment by danimtb** [comment]
A solution for this issue can be found in the #4958 PR but it is still a draft

**Comment by memsharded** [comment]
The PR #4958 has been merged, so this will be released in Conan 1.20.
